packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -83,35 +98,71 @@ const balanceOptions = [
 export const Stackbox = () => {
   const searchTokenBarRef = useRef<HTMLInputElement>(null);
   const [isPickingFromToken, setIsPickingFromToken] = useState<boolean>(false);
-  const [fromToken, setFromToken] = useState<TokenWithBalance | null>();
-  const [toToken, setToToken] = useState<TokenWithBalance | null>();
+
   const { closeModal, isModalOpen, openModal } = useModalContext();
   const {
     selectedStrategy,
     setSelectedStrategy,
     setShouldResetStackbox,
     shouldResetStackbox,
   } = useStrategyContext();
-  const { tokenListWithBalances } = useTokenListContext();
-  const signer = useEthersSigner();
+  const {
+    tokenListWithBalances,
+    getTokenFromList,
+    isLoading: isTokenListLoading,
+  } = useTokenListContext();
 
   const { chain } = useNetwork();
-  const { address, isConnected } = useAccount();
-  const { data: balance } = useBalance({
-    address: Boolean(fromToken) ? address : undefined,
-    token: fromToken?.address as `0x${string}`,
-    chainId: chain?.id,
+  const { switchNetwork } = useSwitchNetwork({
+    onSuccess(data) {
+      setChainId(data.id);
+    },
+  });
+  const { address } = useAccount();
+
+  const TokenParser = createParser({
+    parse: (address: string) => getTokenFromList(address),
+    serialize: (token) => token?.address || "",
   });
-  const [tokenAmount, setTokenAmount] = useState("");
 
-  const [frequency, setFrequency] = useState<FREQUENCY_OPTIONS>(
-    FREQUENCY_OPTIONS.hour
+  const [fromToken, setFromToken] = useQueryState<TokenWithBalance>(
+    "fromToken",
+    TokenParser.withDefault(
+      chain?.id && !chain?.unsupported
+        ? DEFAULT_TOKENS_BY_CHAIN[chain.id].from
+        : DEFAULT_TOKENS_BY_CHAIN[ChainId.GNOSIS].from
+    )
+  );
+  const [toToken, setToToken] = useQueryState<TokenWithBalance>(
+    "toToken",
+    TokenParser.withDefault(
+      chain?.id && !chain?.unsupported
+        ? DEFAULT_TOKENS_BY_CHAIN[chain.id].to
+        : DEFAULT_TOKENS_BY_CHAIN[ChainId.GNOSIS].to
+    )
+  );
+  const [tokenAmount, setTokenAmount] = useQueryState(
+    "tokenAmount",
+    parseAsString.withDefault("")
   );
-  const [startDateTime, setStartDateTime] = useState<Date>(
-    new Date(Date.now())
+  const [frequency, setFrequency] = useQueryState(
```

Comment:
**MINOR**, non-blocking:
Dunno why but the `useQueryState()` is not inferring correctly the type for `frequency`
```suggestion
  const [frequency, setFrequency]: [
    FREQUENCY_OPTIONS,
    (frequency: FREQUENCY_OPTIONS) => {}
  ] = useQueryState(
```

Reply:
It ins't? On my side it is

![Screenshot 2023-10-31 at 11 31 56](https://github.com/SwaprHQ/stackly-ui/assets/23079677/d9ca87c8-fb55-4813-9e95-0a437c13a916)

Reply:
When I was checking the code locally it started to highlight a few places where we were using `frequency` in that file, maybe it was some cached stuff